grabbit/src/main/groovy/com/twcable/grabbit/client/servlets/GrabbitJobServlet.groovy

```diff
@@ -47,59 +48,41 @@ class GrabbitServlet extends SlingAllMethodsServlet {
     @Reference(bind = 'setClientService')
     ClientService clientService
 
-    @Reference(bind = 'setServerService')
-    ServerService serverService
-
 
     @Override
     void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) {
-        final path = request.getParameter("path")
-        final excludePaths = request.getParameterValues("excludePath")
-
-        if (path) { //Server Get Request
-            final decodedPath = URLDecoder.decode(path, "utf-8")
-            final decodedExcludePaths = excludePaths.collect { String ep -> URLDecoder.decode(ep, 'UTF-8') }
-            response.contentType = "application/octet-stream"
-
-            final afterDateString = URLDecoder.decode(request.getParameter("after") ?: "", "utf-8")
-
-            if(afterDateString) {
-                log.info "Path : $decodedPath, Exclude Paths: $decodedExcludePaths, " +
-                        "AfterDate String : $afterDateString. Will send only delta content"
-            }
-
-            //The Login of the user making this request.
-            //This user will be used to connect to JCR
-            //If the User is null, 'anonymous' will be used to connect to JCR
-            final serverUsername = request.remoteUser
-            serverService.getContentForRootPath(serverUsername, decodedPath, decodedExcludePaths ?: null,
-                    afterDateString ?: null, response.outputStream)
-        }
-        else {
-            final jobExecutionId = request.resource.resourceMetadata[JobResource.JOB_EXECUTION_ID] as String ?: ""
+        final jobExecutionId = request.resource.resourceMetadata[JobResource.JOB_EXECUTION_ID_KEY] as String
+        if(jobExecutionId) {
             if (request.pathInfo.endsWith("html")) {
+                response.setStatus(SC_OK)
                 response.writer.write("TODO : This will be replaced by a UI representation for the Jobs Status")
-            }
-            else if (request.pathInfo.endsWith("json")) {
-                final String jsonString = getJsonString(jobExecutionId)
-                log.debug "Current Status : ${jsonString}"
-                response.contentType = "application/json"
-                response.status = HttpServletResponse.SC_OK
-                response.writer.write(jsonString)
-            }
-            else {
-                log.error "Illegal Selector for current request : ${request.pathInfo}"
+                return
+            } else {
+                final String jsonResponse = getJsonResponseForJobStatus(jobExecutionId)
+                if(jsonResponse) {
+                    log.debug "Current Status : ${jsonResponse}"
+                    response.setStatus(SC_OK)
+                    response.contentType = "application/json"
+                    response.writer.write(jsonResponse)
+                    return
+                }
```

Comment:
Yeah, changed things a bit to distinguish between a jobID that doesn't correspond to a job, and a jobID that doesn't exist in the logs/response.